Add callbacks and promises exercises

40thieves opened the pull request:
Adds exercises for callbacks and promises.

I mostly just adapted from them old syllabus, which had them in this [CodePen](https://codepen.io/fortythieves/pen/QOYgWb?editors=1010).

@BexB1 @kkarimi @kabaros

djgrant commented on week-5/B-promises/exercise.js:
```diff
@@ -0,0 +1,156 @@
+  EXERCISE 2
+  =======
+
+  Write a callback function that inserts the result into the #exercise2
```
I think some documentation is needed to explain how this callback gets called.

djgrant replied:
Wasn't clear to me until I scrolled down and looked at the function definition for functionThatCallsBack

djgrant commented on week-5/B-promises/exercise.js:
```diff
@@ -0,0 +1,156 @@
+  EXERCISE 3
+  =======
+
+  Write a function that calls the callback with "Hello from the callback"
```
....with _the parameter_ "Hello from the callback"

40thieves commented:
Thanks for the review 😄

> I think it would be good for them to be tackled in separate files.

Yeah, I was thinking that week5 has a lot of stuff now. But as I mentioned on slack I think we might want to split up this content a bit more (which we should do separately). I've moved the callbacks exercises to their own directory.

I've also addressed your other feedback.

djgrant commented:
Looks good!
On Mon, 11 Jun 2018 at 20:27, Alasdair Smith <notifications@github.com>
wrote:

> Thanks for the review 😄
>
> I think it would be good for them to be tackled in separate files.
>
> Yeah, I was thinking that week5 has a lot of stuff now. But as I mentioned
> on slack I think we might want to split up this content a bit more (which
> we should do separately). I've moved the callbacks exercises to their own
> directory.
>
> I've also addressed your other feedback.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/CodeYourFuture/js-exercises/pull/123#issuecomment-396358224>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABl-9vnUZnJjY60G2ZICLyBrYoR_-tB4ks5t7sSXgaJpZM4Uh30_>
> .
>

40thieves commented:
Cheers! Merging
